File: client/medplanner-app/src/app/models/doctor.ts
import { Clinic } from "./clinic";
import { ClinicDoctor } from "./clinic-doctor";
import { DoctorAdmissionConditions } from "./doctor-admission";
import { Photo } from "./photo";
import { User } from "./user";

export interface Doctor extends User {
    doctorNumber?: string;
    clinicWork?: Clinic[];
    admissionConditions?: DoctorAdmissionConditions[];
    photoId?: string;
    photo?: Photo;
    clinicDoctors?: ClinicDoctor[];
}

File: API/Entities/Doctor.cs

using API.Enums;

namespace API.Entities
{
    public class Doctor : User
    {
        public string? DoctorNumber { get; set; }
        public ICollection<Clinic>? ClinicWork { get; set; }
        public ICollection<DoctorAdmissionConditions>? AdmissionConditions { get; set; }
        public Guid? PhotoId { get; set; }
        public Photo Photo { get; set; }
        public bool IsClinicOwner { get; set; }
        public ICollection<ClinicDoctor>? ClinicDoctors { get; set; }
    }
}
